File: src/Python/82.RemoveDuplicatesFromSortedListII.py

class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None


class Solution:
    def deleteDuplicates(self, head: ListNode) -> ListNode:
        rnode = head
        node = head.next
        lastnode = head
        while node is not None:     
            if lastnode.val == node.val:
                while node is not None:
                    if node.val != lastnode.val:
                        rnode = node
                        lastnode = node
                        rnode = rnode.next
                        break
                    node = node.next
            elif lastnode.val != node.val:
                rnode.next = node
                rnode = rnode.next
                lastnode = node   
            node = node.next
        return rnode


if __name__ == "__main__":
    solution = Solution()
    node = ListNode(0)
    node.next = ListNode(1)
    node.next.next = ListNode(2)
    node.next.next.next = ListNode(2)
    node.next.next.next.next = ListNode(3)
    node.next.next.next.next.next = ListNode(4)
    print(solution.deleteDuplicates(node))
   

File: src/CSharp/1-100/82.RemoveDuplicatesFromSortedListII.cs

using System;
using System.Linq;
using System.Text;
using System.Collections.Generic;


namespace PeefyLeetCode.RemoveDuplicatesFromSortedListII
{

    public class ListNode
    {
        public int val;
        public ListNode next;
        public ListNode(int x) { val = x; }
    }
    public class Solution
    {



        public ListNode DeleteDuplicates(ListNode head)
        {
            if (head == null)
                return head;
            if (head.next != null && head.val == head.next.val)
            {
                while (head.next != null && head.val == head.next.val)
                    head = head.next;
                return DeleteDuplicates(head.next);
            }
            else
            {
                head.next = DeleteDuplicates(head.next);
            }
            return head;
        }
    }
}